### PR TITLE
Merge upstream HEAD(b7bde56d5, 2024-05-14) Stop writing to V2 tables

### PR DESCRIPTION
**Pull request by milo-hyben**

### CODE REVIEW

File: batch/batch/front_end/validate.py [hunk 1]
```diff
@@ -41,86 +41,76 @@
...
-        'port': int_type,
-        required('process'): switch(
-            'type',
-            {
-                'docker': {
-                    required('command'): listof(str_type),
-                    required('image'): image_str,
-                    'mount_docker_socket': bool_type,  # DEPRECATED
-                },
-                'jvm': {
```
Comment (milo-hyben) [inline batch/batch/front_end/validate.py:77]: Applied incoming
	LN 74 new 'jvm'
	'pool_label': str_type, removed

was:
<<<<<<< HEAD
        ),
        'regions': listof(str_type),
        'requester_pays_project': str_type,
        'resources': keyed(
            {
                'memory': anyof(regex(MEMORY_REGEXPAT, MEMORY_REGEX), oneof(*memory_types)),
                'cpu': regex(CPU_REGEXPAT, CPU_REGEX),
                'storage': regex(STORAGE_REGEXPAT, STORAGE_REGEX),
                'machine_type': str_type,
                'pool_label': str_type,
                'preemptible': bool_type,
            }
        ),
        'secrets': listof(
            keyed({required('namespace'): k8s_str, required('name'): k8s_str, required('mount_path'): str_type})
        ),
        'service_account': keyed({required('namespace'): k8s_str, required('name'): k8s_str}),
        'timeout': numeric(**{"x > 0": lambda x: x > 0}),
        'user_code': str_type,
    }
)
=======

File: ci/ci/github.py [hunk 2]
```diff
@@ -873,19 +876,28 @@ async def _update_deploy(self, batch_client, db: Database):
                 else:
                     self.deploy_state = 'failure'
 
+                if not is_test_deployment and self.deploy_state == 'failure':
```
Comment (milo-hyben) [inline ci/ci/github.py:879]: Keeping Both

LN 879:
<<<<<<< HEAD
=======
                if not is_test_deployment and self.deploy_state == 'failure':
                    url = deploy_config.external_url('ci', f'/batches/{self.deploy_batch.id}')
                    deploy_failure_message = f"""
state: {self.deploy_state}
branch: {self.branch.short_str()}
sha: {self.sha}
url: {url}
"""
                    await send_zulip_deploy_failure_message(deploy_failure_message, db, self.sha)
>>>>>>> b7bde56d5

File: batch/batch/front_end/validate.py [hunk 3]
```diff
@@ -41,86 +41,76 @@
...
-    }
-)
-
-batch_update_validator = keyed(
-    {
-        required('token'): str_type,
-        required('n_jobs'): numeric(**{"x > 0": lambda x: isinstance(x, int) and x > 0}),
-    }
-)
+            'jvm': {
```
Comment (milo-hyben) [inline batch/batch/front_end/validate.py:74]: Applied incoming
	LN 74 new 'jvm'
	'pool_label': str_type, removed

was:
```
<<<<<<< HEAD
        ),
        'regions': listof(str_type),
        'requester_pays_project': str_type,
        'resources': keyed(
            {
                'memory': anyof(regex(MEMORY_REGEXPAT, MEMORY_REGEX), oneof(*memory_types)),
                'cpu': regex(CPU_REGEXPAT, CPU_REGEX),
                'storage': regex(STORAGE_REGEXPAT, STORAGE_REGEX),
                'machine_type': str_type,
                'pool_label': str_type,
                'preemptible': bool_type,
            }
        ),
        'secrets': listof(
            keyed({required('namespace'): k8s_str, required('name'): k8s_str, required('mount_path'): str_type})
        ),
        'service_account': keyed({required('namespace'): k8s_str, required('name'): k8s_str}),
        'timeout': numeric(**{"x > 0": lambda x: x > 0}),
        'user_code': str_type,
    }
)
=======
```

File: ci/ci/github.py [hunk 4]
```diff
@@ -978,7 +987,19 @@ async def _start_deploy(self, db: Database, batch_client: BatchClient, steps=DEP
                 },
                 callback=CALLBACK_URL,
             )
-            config.build(deploy_batch, self, scope='deploy')
+            try:
```
Comment (milo-hyben) [inline ci/ci/github.py:990]: Incoming:

```
<<<<<<< HEAD
            config.build(deploy_batch, self, scope='deploy')
=======
            try:
                config.build(deploy_batch, self, scope='deploy')
            except Exception as e:
                deploy_failure_message = f"""
branch: {self.branch.short_str()}
sha: {self.sha}
Deploy config failed to build with exception:
```python
{e}
```
"""
                await send_zulip_deploy_failure_message(deploy_failure_message, db, self.sha)
                raise
>>>>>>> b7bde56d5
```

File: hail/python/hailtop/httpx.py [hunk 5]
```diff
@@ -104,27 +103,33 @@ def __init__(self,
 
         if timeout is None:
             timeout = aiohttp.ClientTimeout(total=20)
```
Comment (milo-hyben) [inline hail/python/hailtop/httpx.py:105]: Combined Both:
```
	<<<<<<< HEAD
            timeout = aiohttp.ClientTimeout(total=20)
=======
            timeout = aiohttp.ClientTimeout(total=5)
        if isinstance(timeout, (float, int)):
            timeout = aiohttp.ClientTimeout(total=timeout)
>>>>>>> b7bde56d5
```

**Comment by milo-hyben** [comment]
Hey @jmarshall, I have merged the first batch. Please let me know if my comments in the conflicts resolution are good enough.

File: batch/batch/front_end/validate.py (hunk 3 above)
Comment (jmarshall) [reply batch/batch/front_end/validate.py:74]: I looked at this one again, and noticed that `pool_label` appears in the `HEAD` part but not the merge-base part. I then remembered a couple of things:

* `pool_label` is one of our CPG-local additions. (Which is something you “just have to know” or can figure it out via the next item)

* A useful trick for resolving these conflicts is to look at what the difference unique to our CPG branch is, i.e., the diff compared to the merge-base, _for this particular file_. So in this case:

```
$ git merge-base 8f6797b03 b7bde56d5
da6668bfd58fe915c54f052844db18975ec7abc1
$ git diff da6668bfd58fe915c54f052844db18975ec7abc1..origin/main -- batch/batch/front_end/validate.py
diff --git a/batch/batch/front_end/validate.py b/batch/batch/front_end/validate.py
index d18dd839f..52bcc405c 100644
--- a/batch/batch/front_end/validate.py
+++ b/batch/batch/front_end/validate.py
@@ -91,6 +91,7 @@ job_validator = keyed(
                 'cpu': regex(CPU_REGEXPAT, CPU_REGEX),
                 'storage': regex(STORAGE_REGEXPAT, STORAGE_REGEX),
                 'machine_type': str_type,
+                'pool_label': str_type,
                 'preemptible': bool_type,
             }
         ),
```

So adding `pool_label` to this list was our local change, and we need to re-add it in the merged version.

That was the correct handling for the `"jvm" part: turns out that was already there, it just got reformatted.

File: ci/ci/github.py (hunk 2 above)
Comment (jmarshall) [reply ci/ci/github.py:879]: Sorry, I should have talked about using `merge.conflictStyle = zdiff3` in your Git config to get the three-part version of these conflicts.

Try the `git diff …… -- ci/ci/github.py` trick for this file and see if you want to change your mind about what happened on these lines :smile:

**Comment by jmarshall** [comment]
Comments are good. A couple of the resolutions need revisiting — largely my fault as there were some tricks I forgot to tell you about, as vaguely mentioned above.

**Comment by milo-hyben** [comment]
Hey @jmarshall , I had fixed those 2 files. Please let me know if any issue.
Thank you!

**Review by jmarshall — APPROVED** [review]

**Pull request merged**